src/main/java/spireMapOverhaul/zones/brokenSpace/BrokenSpaceZone.java

```diff
@@ -0,0 +1,254 @@
+    private final int width, height;
+    private final Color color;
+
+    public BrokenSpaceZone() {
+        this("Broken Space 0", 2, 3);
+
+    }
+
+    private BrokenSpaceZone(String name, int width, int height) {
+        super(ID, Icons.REWARD);
+
+        this.width = width;
+        this.height = height;
+
+        color = Color.WHITE.cpy();
+        this.name = name;
+    }
+
+
+    @Override
+    public void distributeRooms(Random rng, ArrayList<AbstractRoom> roomList) {
+        //Guarantee at least one elite
+        placeRoomRandomly(rng, roomOrDefault(roomList, (room) -> room instanceof MonsterRoomElite, MonsterRoomElite::new));
+
+    }
+
+    @Override
+    public AbstractZone copy() {
+        return new BrokenSpaceZone(name, width, height);
+    }
+
+    @Override
+    public void renderOnMap(SpriteBatch sb, float alpha) {
+        BrokenSpaceRenderPatch.StartFbo(sb);
+        super.renderOnMap(sb, alpha);
+        BrokenSpaceRenderPatch.StopFbo(sb);
+        if (alpha > 0) {
+            FontHelper.renderFontCentered(sb, FontHelper.menuBannerFont, name,
+                    labelX * SPACING_X + OFFSET_X, labelY * Settings.MAP_DST_Y + OFFSET_Y + DungeonMapScreen.offsetY,
+                    Color.WHITE.cpy(), 0.8f
+            );
+        }
+    }
+
+    @Override
+    public boolean generateMapArea(BetterMapGenerator.MapPlanner planner) {
+        return generateNormalArea(planner, width, height);
+    }
+
+    @Override
+    protected boolean canIncludeEarlyRows() {
+        return false;
+    }
+
+    @Override
+    protected boolean canIncludeFinalCampfireRow() {
+        return false;
+    }
+
+    @Override
+    protected boolean canIncludeTreasureRow() {
+        return false;
+    }
+
+    @Override
+    public boolean canSpawn() {
+        return AbstractDungeon.actNum >= 2;
+    }
+
+    @Override
+    public Color getColor() { //I considered changing this to a variable, but a method lets you do funky stuff like a rainbow zone that changes colors or something.
+        return color;
+    }
+
+    public AbstractCard getTrulyRandomCard(Random rng) {
+        AbstractCard.CardRarity rarity = AbstractCard.CardRarity.COMMON;
+
+        // pick RANDOM rarity
+        int roll = rng.random(7);
+        switch (roll) {
+            case 0:
+            case 1:
+                rarity = AbstractCard.CardRarity.COMMON;
+                break;
+            case 2:
+            case 3:
+                rarity = AbstractCard.CardRarity.UNCOMMON;
+                break;
+            case 4:
+            case 5:
+                rarity = AbstractCard.CardRarity.RARE;
+                break;
+            case 6:
+                return AbstractDungeon.returnRandomCurse().makeCopy();
+
+        }
+
+
+        AbstractCard c = CardLibrary.getAnyColorCard(rarity).makeCopy();
+
+        return c;
+
+
+    }
+
+    @Override
+    public ArrayList<AbstractCard> getAdditionalCardReward() {
```

Comment:
Just a heads up, this does not include the upgrade logic from base game so these cards will never be upgraded. Even by the egg relics